driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionDecryptionEventsTest.java

```diff
@@ -0,0 +1,226 @@
+    private ClientEncryption clientEncryption;
+    private Map<String, Map<String, Object>> kmsProviders;
+    private TestCommandListener commandListener;
+    private BsonBinary ciphertext;
+    private BsonBinary malformedCiphertext;
+
+
+    protected abstract MongoClient createMongoClient(MongoClientSettings settings);
+    protected abstract ClientEncryption createClientEncryption(ClientEncryptionSettings settings);
+
+    @BeforeEach
+    public void setUp() {
+        assumeTrue(serverVersionAtLeast(6, 0));
+        assumeFalse(isStandalone());
+        assumeFalse(isServerlessTest());
+
+        getDefaultDatabase().getCollection("decryption_events").drop();
+        getDefaultDatabase().createCollection("decryption_events");
+
+        kmsProviders = new HashMap<>();
+        Map<String, Object> localProviderMap = new HashMap<>();
+        localProviderMap.put("key",
+                Base64.getDecoder().decode(
+                        "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZ"
+                                + "GJkTXVyZG9uSjFk"));
+        kmsProviders.put("local", localProviderMap);
+
+        commandListener = new TestCommandListener();
+        clientEncryption = createClientEncryption(ClientEncryptionSettings.builder()
+                .keyVaultMongoClientSettings(getMongoClientSettingsBuilder().addCommandListener(commandListener).build())
+                .keyVaultNamespace("keyvault.datakeys")
+                .kmsProviders(kmsProviders)
+                .build());
+
+        BsonBinary keyId = clientEncryption.createDataKey("local");
+
+        ciphertext = clientEncryption.encrypt(new BsonString("hello"), new EncryptOptions("AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic").keyId(keyId));
+
+        // Copy ciphertext into a variable named malformedCiphertext. Change the last byte. This will produce an invalid HMAC tag.
+        byte[] malformedBytes = ciphertext.getData().clone();
+        malformedBytes[malformedBytes.length - 1] = (byte) (malformedBytes[malformedBytes.length - 1] == 0 ? 0 : 1);
+        malformedCiphertext = new BsonBinary(ciphertext.getType(), malformedBytes);
+
+        commandListener = new TestCommandListener();
+        encryptedClient = createMongoClient(getMongoClientSettingsBuilder()
+                .autoEncryptionSettings(
+                        AutoEncryptionSettings.builder()
+                                .keyVaultNamespace("keyvault.datakeys")
+                                .kmsProviders(kmsProviders)
+                                .build())
+                .retryReads(false)
+                .addCommandListener(commandListener)
+                .build());
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        if (clientEncryption != null) {
+            clientEncryption.close();
+        }
+        if (encryptedClient != null) {
+            encryptedClient.close();
+        }
```

Comment:
When we close resources this way, we risk to not close some of them if closing a previous resource fails. I propose to use the same approach you used in `Crypt.close`. This is also applicable to `AbstractClientSideEncryptionUniqueIndexKeyAltNamesTest` and `AbstractClientSideEncryptionExplicitEncryptionTest`.

Reply:
Done

Reply:
Without `@SuppressWarnings("try")`, like in `Crypt.close`, the compiler is unhappy and spits out warnings.